### PR TITLE
add ci check

### PR DESCRIPTION
**Pull request by BusyJay**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+language: rust
+os:
+  - linux
+  - osx
+rust:
+  - stable
+  - beta
+  - nightly
+matrix:
+  allow_failures:
```
Comment (overvenus) [inline .travis.yml]: Why allow?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - C:\ProgramData\chocolatey\lib -> appveyor.yml
+
+environment:
+  TARGET: x86_64-pc-windows-msvc
+
+install:
+  - ps: Start-FileDownload "https://static.rust-lang.org/dist/rust-nightly-${env:TARGET}.exe"
+  - rust-nightly-%TARGET%.exe /VERYSILENT /NORESTART /DIR="C:\Program Files (x86)\Rust"
+  - SET PATH=%PATH%;C:\Program Files (x86)\Rust\bin
+  - rustc -V
```
Comment (overvenus) [inline appveyor.yml]: Why print version?

File: .travis.yml (hunk 1 above)
Comment (BusyJay) [reply .travis.yml]: Because clippy may fail to compile.

File: appveyor.yml (hunk 2 above)
Comment (BusyJay) [reply appveyor.yml]: In case of failure, version information is helpful.

Comment (overvenus) [reply appveyor.yml]: So how about add it to linux ci too?

Comment (BusyJay) [reply appveyor.yml]: Travis CI will print it automatically.

**Comment by BusyJay** [comment]
ping

**Comment by siddontang** [comment]
PTAL @overvenus

**Pull request merged**